src/vcpkg/commands.build.cpp

```diff
@@ -1065,274 +1066,6 @@ namespace vcpkg
-                if (auto hash = maybe_hash.get())
-                {
-                    return std::move(*hash);
-                }
-                else
-                {
-                    return nullopt;
-                }
-            });
-
-            if (auto header_hash = maybe_header_hash.get())
-            {
-                return *header_hash;
-            }
-        }
-
-        return "none";
-    }
-
-    static void abi_entries_from_pre_build_info(const Filesystem& fs,
-                                                Cache<Path, Optional<std::string>>& grdk_cache,
-                                                const PreBuildInfo& pre_build_info,
-                                                std::vector<AbiEntry>& abi_tag_entries)
-    {
-        if (pre_build_info.public_abi_override)
-        {
-            abi_tag_entries.emplace_back(
-                "public_abi_override",
-                Hash::get_string_hash(pre_build_info.public_abi_override.value_or_exit(VCPKG_LINE_INFO),
-                                      Hash::Algorithm::Sha256));
-        }
-
-        for (const auto& env_var : pre_build_info.passthrough_env_vars_tracked)
-        {
-            if (auto e = get_environment_variable(env_var))
-            {
-                abi_tag_entries.emplace_back(
-                    "ENV:" + env_var, Hash::get_string_hash(e.value_or_exit(VCPKG_LINE_INFO), Hash::Algorithm::Sha256));
-            }
-        }
-
-        if (pre_build_info.target_is_xbox)
-        {
-            abi_tag_entries.emplace_back("grdk.h", grdk_hash(fs, grdk_cache, pre_build_info));
-        }
-    }
-
-    static void populate_abi_tag(const VcpkgPaths& paths,
-                                 InstallPlanAction& action,
-                                 std::unique_ptr<PreBuildInfo>&& proto_pre_build_info,
-                                 Span<const AbiEntry> dependency_abis,
-                                 Cache<Path, Optional<std::string>>& grdk_cache)
-    {
-        Checks::check_exit(VCPKG_LINE_INFO, static_cast<bool>(proto_pre_build_info));
-        const auto& pre_build_info = *proto_pre_build_info;
-        const auto& toolset = paths.get_toolset(pre_build_info);
-        auto& abi_info = action.abi_info.emplace();
-        abi_info.pre_build_info = std::move(proto_pre_build_info);
-        abi_info.toolset.emplace(toolset);
-
-        if (action.build_options.use_head_version == UseHeadVersion::YES)
-        {
-            Debug::print("Binary caching for package ", action.spec, " is disabled due to --head\n");
-            return;
-        }
-        if (action.build_options.editable == Editable::YES)
-        {
-            Debug::print("Binary caching for package ", action.spec, " is disabled due to --editable\n");
-            return;
-        }
-
-        abi_info.compiler_info = paths.get_compiler_info(*abi_info.pre_build_info, toolset);
```

Comment:
Why do we not need compiler info if either `--head` or `--editable` were used?

Reply:
If you don't have a abi_info object you don't need the fill the compiler_info entry in it.